ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_statistical.py

```diff
@@ -74,6 +74,84 @@ def statistical_dtype_values(draw, *, function):
     return dtype, values, axis
 
 
+# TODO: numpy does not support bfloat16
+# TODO: put this function inside statistical_dtype_values if needed
+@st.composite
+def _statistical_dtype_xs_bins_range_axis_castable(draw, n: int = 2):
+    available_dtypes = draw(helpers.get_dtypes(kind="float"))
+    if "bfloat16" in available_dtypes:
+        available_dtypes.remove("bfloat16")
+    shape = draw(helpers.get_shape(min_num_dims=1, min_dim_size=2))
+    dtype, values = draw(
+        helpers.dtype_and_values(
+            available_dtypes=available_dtypes,
+            num_arrays=n,
+            large_abs_safety_factor=2,
+            small_abs_safety_factor=2,
+            safety_factor_scale="log",
+            shape=shape,
+            shared_dtype=True,
+        )
+    )
+    axis = draw(helpers.get_axis(shape=shape, force_int=True))
+    dtype1, dtype2 = draw(helpers.get_castable_dtype(available_dtypes, dtype[0]))
+    bins = draw(
+        helpers.array_values(
+            dtype=dtype1, min_value=0, shape=draw(helpers.get_shape(max_num_dims=1))
+        )
+    )
+    range = (0, 0)
+    if bins.size == 1:
+        while range[0] == range[1]:
+            range = (
+                draw(
+                    st.floats(
+                        allow_nan=False, allow_infinity=False, allow_subnormal=False
+                    )
+                ),
+                draw(
+                    st.floats(
+                        allow_nan=False, allow_infinity=False, allow_subnormal=False
+                    )
+                ),
+            )
+        range = sorted(range)
+        return dtype1, values, int(bins), range, axis, dtype2
+    else:
+        bins = sorted(bins)
+        range = None
+        return dtype1, values, bins, range, axis, dtype2
+
+
+@handle_test(
+    fn_tree="functional.ivy.experimental.histogram",
+    private_data_generated=_statistical_dtype_xs_bins_range_axis_castable(),
+    extend_lower_interval=st.booleans(),
+    extend_upper_interval=st.booleans(),
+    density=st.booleans(),
+)
+def test_histogram(
```

Comment:
Hi! Hope you're doing well! I'm adding this comment to "add a review" so it's easier for us to know when you finished the changes I asked you for 😊

Reply:
Hi @ZoeCD! I could solve the problem and other errors that came after, but I got stuck now, because the test function is consuming too much memory and no errors are detected. I've tried looking for help in discord, but I still couldn't figure out what's the matter with my solution. Could you you help, please? I've pushed the last modification to my fork.

Reply:
Hi! Could you please share what message you are getting? Is it specifically from this test function or is it happening in other tests too?

Reply:
Hi @ZoeCD! The test_histogram function runs into this error:

```
/usr/bin/python3 /opt/.pycharm_helpers/pycharm/_jb_pytest_runner.py --target test_statistical.py::test_histogram 
Testing started at 14:51 ...
Launching pytest with arguments test_statistical.py::test_histogram --no-header --no-summary -q in /ivy/ivy_tests/test_ivy/test_functional/test_experimental/test_core

Use Database in ReadOnly Mode with local caching !
============================= test session starts ==============================
collecting ... collected 4 items

test_statistical.py::test_histogram[cpu-ivy.functional.backends.numpy-False-False] 
Process finished with exit code 137 (interrupted by signal 9: SIGKILL)
```

When I checked the memory usage, it was consuming all 32GB I have on my computer. 

I can't confirm this error isn't happening with other experimental test functions from the statistical folder because they are getting errors that need to be corrected before running the test, like some fixture errors.

Reply:
@ZoeCD I've pushed the last version I tried to run.